src/cardSystem/Entity.py

```diff
@@ -2,29 +2,85 @@
-    def __init__(self, name, image=None):
+    def __init__(self, name, health = 10, image=None):
+        # For Render
         self.name = name
-        self.field_index = None  # Keep track of which field it is on
-        self.index = None
+        self.fieldTile_index = None  # Keep track of which field it is on
+        self.image = image
+
+        # Deck & Card
         self.deck = []
-        self.hand = []
-        self.selected_card = None
+        self.cardsOnHand = []
+        self.selectedCard = None
+
+        # Entity Stats
+        self.health = health
+        self.attack = 0
+        self.defense = 0
+        self.speed = 0
+        self.range = 0
         self.stunt = False
-        self.image = image
+        self.buffs = [] # list of buff (or debuff?) apply on entity
 
-    def move_to(self, field, fields):
-        if field.is_occupied():  # Check if the field is occupied
-            print("Field is already occupied!")
+    def print_stats(self):
+        print(f'{self.name} stats - HP: {self.health}, ATK: {self.attack}, DEF: {self.defense}, SPD: {self.speed}, RNG: {self.range}')
+
+    def move_to(self, fieldTile, field):
+        if fieldTile.is_occupied():  # Check if the fieldTile is occupied
+            print("fieldTile is already occupied!")
             return
 
-        # Remove from the current field if necessary
-        if self.field_index is not None:
-            current_field = fields[self.field_index]  # Access fields from the passed list
-            current_field.remove_entity()  # Remove from current field
+        # Remove from the current fieldTile if necessary
+        if self.fieldTile_index is not None:
+            field[self.fieldTile_index].remove_entity()  # Remove from current fieldTile
+            
+        fieldTile.place_entity(self)  # Place the entity in the new fieldTile
+        self.fieldTile_index = fieldTile.index  # Update the fieldTile index
 
-        field.place_entity(self)  # Place the entity in the new field
+    def add_buff(self, buff):
+        self.buffs.append(buff)
+    
+    def apply_existing_buffs(self):
+        # self.print_stats()
+        for buff in self.buffs:
+            buff.apply(self)
+        # self.print_stats()
+
+    def select_card(self, card):
+        print(f'\t{self.name} selected card: {card.name}')
+        self.selectedCard = card
+        self.selectedCard.isSelected = True
+
+        self.attack = card.attack
+        self.defense = card.defense
+        self.speed = card.speed
+        self.range = card.range
+
+    def reset(self):
+        # remove selected card and draw new card
+        self.cardsOnHand.remove(self.selectedCard)
+        self.selectedCard = None
+        self.cardsOnHand.append(self.deck.draw(1)[0])
+
+        # reset stats
+        self.attack = 0
+        self.defense = 0
+        self.speed = 0
+        self.range = 0
+
+        # reset buffs
+        for buff in self.buffs:
+            buff.turn_passed()
```

Comment:
I don't think `reset` method would be the naming of the passing turn. Maybe we used another words to indicate it to reduce confusion?

Reply:
agree, I will change to `turn_passed`

Reply:
For clarity, I think i will change to `next_turn` instead.